File: java-to-kotlin/src/main/java/nl/rabobank/kotlinmovement/recipes/data/RecipesEntity.java
package nl.rabobank.kotlinmovement.recipes.data;


import jakarta.persistence.*;
import lombok.*;

import java.util.Collections;
import java.util.Set;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Table(name = "recipes")
public final class RecipesEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long recipeId;
    @NonNull
    private String recipeName;
    @OneToMany(mappedBy = "recipes", fetch = FetchType.LAZY, cascade = CascadeType.ALL, orphanRemoval = true)
    private Set<IngredientsEntity> ingredients = Collections.emptySet();
}

File: java-to-kotlin-complete/src/main/kotlin/nl/alimeshkat/recipes/data/RecipesEntity.kt
package nl.alimeshkat.recipes.data

import jakarta.persistence.*

@Entity
@Table(name = "recipes")
class RecipesEntity(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val recipeId: Long? = null,
    val recipeName: String,
    @OneToMany(mappedBy = "recipes", fetch = FetchType.LAZY, cascade = [CascadeType.ALL], orphanRemoval = true)
    val ingredients: Set<IngredientsEntity>? = emptySet()
)
